napari/_vispy/layers/base.py

```diff
@@ -122,6 +125,33 @@ def _on_blending_change(self):
         self.node.set_gl_state(**blending_kwargs)
         self.node.update()
 
+    def _on_overlays_change(self):
+        # avoid circular import; TODO: fix?
+        from napari._vispy.utils.visual import create_vispy_overlay
+
+        overlay_models = self.layer._overlays.values()
+
+        for overlay in overlay_models:
+            if overlay not in self.overlays:
+                overlay_visual = create_vispy_overlay(
```

Comment:
This seems to imply that an overlay is associated with a specific layer instance, which seems weird for canvas overlays (e.g. scale bar). Is it actually possible to have an instance of `CanvasOverlay` here?

Reply:
The distinction between canvas overlays and scene overlay is no longer tied to viewer vs layer like it was at the beginning of this PR; they are now two orthogonal concepts:

> Overlays can differ on 2 axes which can be combined in any way:
> 
> - canvas/scene: canvas overlays live in canvas space (2D, unaffected by world transforms), while scene overlays live in scene space (2D or 3D, "displayed world")
> - viewer/layer: viewer overlays (scale_bar, text, axes) live on the ViewerModel, while layer overlays (bounding_box, transform_box, selection_box) live on each Layer.

Currently, there are no `CanvasOverlays` that are also `LayerOverlays`, but in discussions we envisioned several (for example, a in-canvas colorbar showing the colormap). Now that I think about it, we probably want the `SelectionBox` to also live in canvas space (so we can use it both in 2D and 3D). So you can have all combinations:
- viewer + canvas: `ScaleBar`
- viewer + scene: `Axes`
- layer + canvas: `ColorBar`, `SelectionBox`
- layer + scene: `BoundingBox`

Reply:
Ah, I do really want a layer specific colorbar, so this explanation makes a lot of sense. Thanks for clarifying!